build.gradle

```diff
@@ -1,18 +1,26 @@
 plugins {
+  id 'org.jetbrains.kotlin.jvm' version '1.6.21' apply false
   id 'com.gradle.plugin-publish' version '0.21.0'
-  id 'java-gradle-plugin'
   id 'maven-publish'
-  id 'codenarc'
+  id 'java-gradle-plugin'
+  id 'java-library'
   id 'groovy'
-}
-
-repositories {
-  mavenCentral()
+  id 'codenarc'
 }
 
 group = GROUP
 version = VERSION_NAME
 
+configurations.configureEach {
+  resolutionStrategy {
+    eachDependency { details ->
+      if (details.requested.group == 'org.jetbrains.kotlin') {
+        details.useVersion '1.6.21'
+      }
```

Comment:
You mean add Kotlin dependency to the `implementation` for both modules?

Reply:
I think you can use `all` but something like that. Maybe actually what you have is correct, but add `details.requested.version < 1.6.21` or something similar. I'm just guessing that your actual desire is for this to be a minimum version rather than forced despite upgrade options.